src/helpers/prf_compute/multiplicative_inverse.rs

```diff
@@ -0,0 +1,170 @@
+use std::cmp;
+
+use num_bigint::{BigInt, BigUint, ToBigInt};
+use num_integer::Integer;
+use num_traits::{identities::Zero, One, Signed};
+
+fn extended_binary_gcd(m: &BigUint, n: &BigUint) -> (BigInt, BigInt, BigUint) {
+    #[inline]
+    fn count_twos_multiple(x: &BigUint) -> u64 {
+        x.trailing_zeros().unwrap_or(0)
+    }
+
+    // Assume non zero
+    assert!(!m.is_zero());
+    assert!(!n.is_zero());
+
+    let mut x_ebgcd = m.clone();
+    let mut y_ebgcd = n.clone();
+    let mut g_ebgcd = BigUint::one();
+
+    // find common factors of 2
+    let shift = cmp::min(count_twos_multiple(&x_ebgcd), count_twos_multiple(&y_ebgcd));
```

Comment:
This seems like a funny optimization to throw in. I wouldn't anticipate this being a really common use-case in IPA given that we are finding the GCD of a number and a *prime*... In fact, I'm almost completely confident that we will never shift due to that...

...let's remove this code.

Reply:
True.. makes sense